### PR TITLE
Issue 200: Update k8 and Controller-run time version

### LINKED ISSUE #200: Update k8 and Controller-run time versions
### Description

Currently the controller-run time version used in operator is 0.6.5 and K8 version is 1.19. Update these to latest versions. Also remove `replace` directives from `go.mod` 
### Importance

 should-have

### Location

go.mod
Also update affected modules

### Suggestions for an improvement

upgrade version of controller-runtime to `0.11.0`

### PR DESCRIPTION
**Pull request by anishakj**
### Change log description

Updating controller-runtime to `11.0` and k8 version to `23.0`

### Purpose of the change

Fixes #200

### What the code does

Updated the versions in `go.mod` and removed replace directives
Changed the Api calls accordance with new controller run time
Rewrite the e2e tests using `envtest`

### How to verify it

All unit and e2e tests should pass

### CODE REVIEW

**Comment by AndreyAdnreyev** [comment]
LGTM

File: Dockerfile [hunk 1]
```diff
@@ -23,16 +23,20 @@ ARG GIT_SHA=0000000
...
 
 COPY pkg ./pkg
-COPY cmd ./cmd
 COPY go.mod ./
 COPY go.sum ./
 
 # Download all dependencies.
-RUN go mod download
+RUN go mod download -x
+
```
Comment (nishant-yt) [inline Dockerfile:31]: Nit : To avoid bloating of console with download messages , avoid using `-x`

Comment (anishakj) [reply Dockerfile:31]: Done

**Review by nishant-yt — APPROVED** [review]
LGTM

**Pull request merged**